src/docker_hub.ml

```diff
@@ -350,15 +358,19 @@ module Config = struct
     Json.pp fmt json
 end
 
+external reraise : exn -> 'a = "%reraise"
+
 let fetch_rootfs ~output_file {Manifest.rootfs_digest; _} {Token.token; name; _} =
-  let ( >>= ) = Lwt.bind in
-  hurl ~meth:`GET
-    ~headers:[("Accept", Manifest.rootfs_media_type); ("Authorization", fmt "Bearer %s" token)]
-    (fmt "https://registry-1.docker.io/v2/%s/blobs/%s" name rootfs_digest)
-  >>= function
+  match
+    hurl ~meth:`GET
+      ~headers:[("Accept", Manifest.rootfs_media_type); ("Authorization", fmt "Bearer %s" token)]
+      (fmt "https://registry-1.docker.io/v2/%s/blobs/%s" name rootfs_digest)
+  with
   | Ok x ->
-      Lwt_io.with_file ~mode:Lwt_io.Output (Fpath.to_string output_file) begin fun ch ->
-        Lwt_io.write ch x >|= fun () ->
-        Ok ()
-      end
-  | Error e -> Lwt.return (Error e)
+      let fd = Unix.openfile (Fpath.to_string output_file) Unix.[O_WRONLY; O_TRUNC; O_CREAT] 0o666 in
+      let fd = try Miou_unix.of_file_descr fd with e -> Unix.close fd; reraise e in
+      Fun.protect ~finally:(fun () -> Miou_unix.close fd) begin fun () ->
+        Miou_unix.write fd x;
+      end;
```

Comment:
I feel like there is a simpler way to do that. Could `Miou_unix` provide an `openfile` function or would it be more something for a higher level library? If it's the latter, is that already planned? If not i'm happy to start one